services/QuillLMS/app/controllers/activity_sessions_controller.rb

```diff
@@ -56,6 +54,17 @@ def activity_session_from_classroom_unit_and_activity
     redirect_to "/activity_sessions/#{started_activity_session_id}/play"
   end
 
+  private def set_variables_and_kick_off_worker_for_partner_session
```

Comment:
I like this extraction to cut down on complexity.  You could go refactor further and pull out some assignments (which I'm noticing I missed when I did the determine_layout method below).
```
def result
  ...
  run_partner_session_worker
  ...
end

private def partner_session? = session[:partner_session].present? && partner_name && partner_session_id
private def partner_name = session[:partner_session]["partner_name"]
private def partner_session_id = session[:partner_session]["session_id"]

private def run_partner_session_worker
  return unless partner_session?

  activity_url = @activity_session.activity.anonymous_module_url.to_s
  results_url = url_for(action: 'result', uid: @activity_session.uid)
  
  AmplifyReportActivityWorker.perform_async(
    partner_session_id, 
    @activity_session.activity.name, 
    @activity_session.activity.description, 
    @activity_session.percentage, 
    activity_url, 
    results_url
  )
end
```

Then at the bottom refactor determine layout:
```
  private def determine_layout = "integrations" if partner_session?
```